lib/matplotlib/colorbar.py

```diff
@@ -222,102 +224,69 @@ def _set_ticks_on_axis_warn(*args, **kw):
-
-    def tick_values(self, vmin, vmax):
-        # flip if needed:
-        if vmin > vmax:
-            vmin, vmax = vmax, vmin
-        vmin = max(vmin, self._colorbar.norm.vmin)
-        vmax = min(vmax, self._colorbar.norm.vmax)
-        ticks = super().tick_values(vmin, vmax)
-        rtol = (vmax - vmin) * 1e-10
-        return ticks[(ticks >= vmin - rtol) & (ticks <= vmax + rtol)]
-
-
-class _ColorbarAutoMinorLocator(ticker.AutoMinorLocator):
-    """
-    AutoMinorLocator for Colorbar
-
-    This locator is just a `.AutoMinorLocator` except the min and max are
-    clipped by the norm's min and max (i.e. vmin/vmax from the
-    image/pcolor/contour object).  This is necessary so that the minorticks
-    don't extrude into the "extend regions".
-    """
-
-    def __init__(self, colorbar, n=None):
-        """
-        This ticker needs to know the *colorbar* so that it can access
-        its *vmin* and *vmax*.
+        Parameters
+        ----------
+        parent : Axes
+            Axes that specifies the position of the colorbar.
+        userax : boolean
+            True if the user passed `.Figure.colorbar` the axes manually.
         """
-        self._colorbar = colorbar
-        self.ndivs = n
-        super().__init__(n=None)
-
-    def __call__(self):
-        vmin = self._colorbar.norm.vmin
-        vmax = self._colorbar.norm.vmax
-        ticks = super().__call__()
-        rtol = (vmax - vmin) * 1e-10
-        return ticks[(ticks >= vmin - rtol) & (ticks <= vmax + rtol)]
-
-
-class _ColorbarLogLocator(ticker.LogLocator):
-    """
-    LogLocator for Colorbarbar
-
-    This locator is just a `.LogLocator` except the min and max are
-    clipped by the norm's min and max (i.e. vmin/vmax from the
-    image/pcolor/contour object).  This is necessary so ticks don't
-    extrude into the "extend regions".
 
-    """
-    def __init__(self, colorbar, *args, **kwargs):
+        if userax:
+            # copy position:
+            fig = parent.figure
+            parent_ax = fig.add_axes(parent.get_position())
+            # copy the locator if one exists:
+            parent_ax._axes_locator = parent._axes_locator
+            # remove the parent from the figure...
+            parent.remove()
+        else:
+            parent_ax = parent
+
+        inner_ax = parent_ax.inset_axes([0, 0, 1, 1])
+        self.__dict__ = inner_ax.__dict__
+
+        self.parent_ax = parent_ax
+        self.inner_ax = inner_ax
+        self.parent_ax.xaxis.set_visible(False)
+        self.parent_ax.yaxis.set_visible(False)
+        self.parent_ax.set_facecolor('none')
+        self.parent_ax.tick_params = self.inner_ax.tick_params
+        self.parent_ax.set_xticks = self.inner_ax.set_xticks
+        self.parent_ax.set_yticks = self.inner_ax.set_yticks
+        for attr in ["get_position", "set_position", "set_aspect"]:
+            setattr(self, attr, getattr(self.parent_ax, attr))
+        if userax:
```

Comment:
I don't think this if statement is necessary? The other branch of this would just put this dict on a newly created axes.

Reply:
This is necessary if we want the user's parent axes to be as much like the new axes as possible (short of being the same object).  This is what locks out all the methods on the parent axes (not outer!) and points them at the inner axes.  That way if the user does `parent.set_xscale('log')` it is the same as `cb.ax.set_xscale('log')`sa.

Reply:
I still think you could do that to `outer_ax` regardless of whether a `userax` was passed in or not? In the other case  it is just a new axis you've created so I don't think it hurts anything to assign the dict in that case.

I also think you missed the argument `parent` change here too.